src/helpers/validate.ts

```diff
@@ -1,12 +1,13 @@
 import { isEntityId, isEntityPageTitle, isPropertyId, isRevisionId } from './helpers.js'
 
-const validate = (name, testFn) => value => {
-  if (!testFn(value)) throw new Error(`invalid ${name}: ${value}`)
+function validate<T> (name: string, testFn: (value: unknown) => value is T) {
+  return function (value: unknown): value is T {
+    if (!testFn(value)) throw new Error(`invalid ${name}: ${value}`)
+    return true
```

Comment:
why does it need to return `true`?

Reply:
I changed the return to `value is T`. But that was kinda unnecessary.
Now I changed the validate method to both ensure TypeScript and runtime checks in 10513ea.

I knew passing functions directly to map, filter, forEach, … is bad as these functions also get more than the first argument which could end up in bugs.
TIL: TypeScript doesn't seem to notice the wrong types then. So I changed it to `o => function(o)` and then TypeScript notices.

Reply:
Thinking about it later: I think we should keep the value is T in there if these methods are relevant for library users.

When they just are internal, keep them simple.
If they should also be used by others, having that type predicate would be helpful as it throws otherwise anyway.